### ISSUE TITLE
Different order of module type resolution checks than video.js

### ISSUE THREAD

**Issue by dbettini** [videojs/videojs-youtube#378]
Hi,
The order of module type resolution should be changed so it favors CommonJS instead of AMD, because of how webpack handles it, as seen in https://github.com/webpack/webpack/issues/919

Webpack always chooses the first type in the compile step, and sets the condition to true. That means that, at the moment, AMD is always chosen for webpack users. It doesn't affect any build at the moment, because the AMD and CommonJS modules are both named 'video.js'. The problem will appear after #373 gets merged, since it will then throw "module not found" error in the compile step. 

There is a way to go around this for webpack users, using the [imports-loader](https://github.com/webpack/imports-loader) to disable AMD completely if you don't plan to use it, by adding this to the module.loaders list in webpack.config:
`{test: /\.js$/, loader: "imports-loader?define=>false"}`

However, the reason why I think this should still be changed is because it is a tech for video.js, and video.js also prioritizes CommonJS (not in the src, but it is changed after the build step).
In other words, it should follow the same rules as video.js. Then, if people want to use webpack+AMD, they will have to use imports-loader anyway (setting require=>false), because of video.js prioritizing CommonJS. If they don't use AMD, they don't have to use imports-loader.

**Comment by gkatsev** [comment]
As [I mentioned](https://github.com/eXon/videojs-youtube/issues/373#issuecomment-218290382) in #373, I think we should remove the custom UMD wrapper we're using in favor of using commonjs (or es6) directly and build a UMD standalone file for requirejs and global using browserify. This way both browserify and webpack users can just require videojs-youtube easily and requirejs and script users can load in the dist file.
Your fix in #379 makes sense as a stopgap until such a thing happens.

**Comment by gkatsev** [comment]
Fixed by #379. Released as part of 2.1.0.
